### PR TITLE
Jvenstad/submit clis

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review. This is WIP, but would be nice to be able to test it in CD, which requires a merge.

### CODE REVIEW

**Comment by jonmv** [comment]
Ah, the first 6 commits are already part of #9270

**Comment by bjorncs** [comment]
> Ah, the first 6 commits are already part of #9270
`git fetch && git rebase origin/master && git push -f` please :)

**Comment by jonmv** [comment]
Think I needed the other one merged first, but can't remember. Anyway, done.

File: hosted-api/src/main/java/ai/vespa/hosted/api/Submission.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        this.applicationTestZip = applicationTestZip;
+    }
+
+    public String repository() { return repository; }
+    public String branch() { return branch; }
+    public String commit() { return commit; }
+    public String authorEmail() { return authorEmail; }
+    public Path applicationZip() { return applicationZip; }
+    public Path applicationTestZip() { return applicationTestZip; }
+
```
Comment (bjorncs) [inline hosted-api/src/main/java/ai/vespa/hosted/api/Submission.java]: Consider adding `equals`, `hashCode` and `toString` as this is a data class.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**